A/AMReX/build_tarballs.jl

```diff
@@ -0,0 +1,79 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "AMReX"
+version = v"21.4.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/AMReX-Codes/amrex/releases/download/21.04/amrex-21.04.tar.gz", "1c610e4b0800b16f7f1da74193ff11af0abfb12198b36a7e565a6a7f793087fa")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd amrex
+mkdir build
+cd build
+if [[ "$target" == *-x86_64-w64-mingw32 ]]; then
+    # TODO: This case should not be necessary
+    mpiopts="-DMPI_HOME=$WORKSPACE/destdir -DMPI_GUESS_LIBRARY_NAME=MSMPI -DMPI_C_LIBRARIES=msmpi64 -DMPI_CXX_LIBRARIES=msmpi64"
+elif [[ "$target" == *-mingw* ]]; then
+    mpiopts="-DMPI_HOME=$WORKSPACE/destdir -DMPI_GUESS_LIBRARY_NAME=MSMPI -DMPI_C_LIBRARIES=msmpi64"
+else
+    mpiopts=
+fi
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN%.*}_gcc.cmake -DCMAKE_BUILD_TYPE=Release -DAMReX_FORTRAN=OFF -DAMReX_OMP=ON -DAMReX_PARTICLES=ON -DBUILD_SHARED_LIBS=ON ${mpiopts} ..
+make -j$(nproc)
+make -j$(nproc) install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# - We can't build with musl since AMReX requires the `fegetexcept` GNU API
+# - We might not want to build `cxxstring_abi="cxx03"` since AMReX requires at least C++14 (C++17 on Windows)
+platforms = [
+    Platform("aarch64", "linux"; cxxstring_abi="cxx03", libc="glibc"),
+    Platform("aarch64", "linux"; cxxstring_abi="cxx11", libc="glibc"),
+    Platform("armv7l", "linux"; call_abi="eabihf", cxxstring_abi="cxx03", libc="glibc"),
+    Platform("armv7l", "linux"; call_abi="eabihf", cxxstring_abi="cxx11", libc="glibc"),
+    Platform("armv7l", "linux"; cxxstring_abi="cxx03", libc="glibc"),
+    Platform("armv7l", "linux"; cxxstring_abi="cxx11", libc="glibc"),
+    Platform("i686", "linux"; cxxstring_abi="cxx03", libc = "glibc"),
+    Platform("i686", "linux"; cxxstring_abi="cxx11", libc = "glibc"),
+    Platform("i686", "windows"; cxxstring_abi="cxx03"),
+    Platform("i686", "windows"; cxxstring_abi="cxx11"),
+    Platform("powerpc64le", "linux"; cxxstring_abi="cxx03", libc="glibc"),
+    Platform("powerpc64le", "linux"; cxxstring_abi="cxx11", libc="glibc"),
+    Platform("x86_64", "freebsd"; cxxstring_abi="cxx03"),
+    Platform("x86_64", "freebsd"; cxxstring_abi="cxx11"),
+    Platform("x86_64", "linux"; cxxstring_abi="cxx03", libc="glibc"),
+    Platform("x86_64", "linux"; cxxstring_abi="cxx11", libc="glibc"),
+    Platform("x86_64", "macos"; cxxstring_abi="cxx03"),
+    Platform("x86_64", "macos"; cxxstring_abi="cxx11"),
```

Comment:
FreeBSD and macOS don't need to specify the C++ string ABI.  I'd suggest to list the base (without `cxxstring_abi`) platforms here and then do
```julia
platforms = expand_cxxstring_abis(platform)
```
which does the right thing:tm:

Reply:
Question: Should I actually build the C++03 ABI? The code requires at least C++14.

Reply:
Yes, that doesn't have to do with the standard of the source code: https://docs.binarybuilder.org/stable/building/#Expanding-C-string-ABIs-or-libgfortran-versions  It's a GCC ABI incompatibility.  While rare (especially if using the official binaries which ship libstdc++ with C++11 string ABI), there might be users around with a libstdc++ with C++03 string ABI

Reply:
And the fact the C++ string ABI is exclusively a GCC thingy is the reason why this isn't a problem on macOS and FreeBSD: we use Clang by default on those platforms.

Reply:
Actually, this package uses GCC on macOS because it requires OpenMP. I assume that GCC there still uses the Clang C++ library?